samples/net/secure_mqtt_sensor_actuator does not compile

### Describe the bug

Demo application created via https://github.com/zephyrproject-rtos/zephyr/pull/70918 uses adi_eval_adin1110ebz as a reference platform, but compilation is failing as reported via https://lists.zephyrproject.org/g/users/topic/117266351

By bisecting from the very first commit, I was not able to compile as well.

### Regression

- [ ] This is a regression.

### Steps to reproduce

$ west build -b adi_eval_adin1110ebz samples/net/secure_mqtt_sensor_actuator --pristine

### Relevant log output

Full logs can be seen in See https://lists.zephyrproject.org/g/users/topic/117266351

```shell
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:30:22: error: array type has incomplete element type 'struct pollfd'
   30 | static struct pollfd fds[1];
      |                      ^~~
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c: In function 'prepare_fds':
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:83:25: error: 'POLLIN' undeclared (first use in this function)
   83 |         fds[0].events = POLLIN;
      |                         ^~~~~~
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:83:25: note: each undeclared identifier is reported only once for each function it appears in
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c: In function 'poll_mqtt_socket':
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:258:14: warning: implicit declaration of function 'poll' [-Wimplicit-function-declaration]
  258 |         rc = poll(fds, nfds, timeout);
      |              ^~~~
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c: In function 'app_mqtt_process':
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:365:38: error: 'POLLIN' undeclared (first use in this function)
  365 |                 if (fds[0].revents & POLLIN) {
      |                                      ^~~~~~
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:373:47: error: 'POLLHUP' undeclared (first use in this function)
  373 |                         if (fds[0].revents & (POLLHUP | POLLERR)) {
      |                                               ^~~~~~~
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:373:57: error: 'POLLERR' undeclared (first use in this function)
  373 |                         if (fds[0].revents & (POLLHUP | POLLERR)) {
      |                                                         ^~~~~~~
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c: In function 'app_mqtt_init':
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:441:22: error: variable 'hints' has initializer but incomplete type
  441 |         const struct addrinfo hints = {
      |                      ^~~~~~~~
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:442:18: error: 'const struct addrinfo' has no member named 'ai_family'
  442 |                 .ai_family = AF_INET,
      |                  ^~~~~~~~~
In file included from T:/Dev/zephyrproject/zephyr/include/zephyr/net/socket.h:33,
                 from T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:11:
T:/Dev/zephyrproject/zephyr/include/zephyr/net/net_ip.h:45:25: warning: excess elements in struct initializer
   45 | #define NET_PF_INET     1            /**< IP protocol family version 4. */
      |                         ^
T:/Dev/zephyrproject/zephyr/include/zephyr/net/net_compat.h:51:24: note: in expansion of macro 'NET_PF_INET'
   51 | #define AF_INET        NET_PF_INET
      |                        ^~~~~~~~~~~
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:442:30: note: in expansion of macro 'AF_INE'
  442 |                 .ai_family = AF_INET,
      |                              ^~~~~~~
T:/Dev/zephyrproject/zephyr/include/zephyr/net/net_ip.h:45:25: note: (near initialization for 'hints')
   45 | #define NET_PF_INET     1            /**< IP protocol family version 4. */
      |                         ^
T:/Dev/zephyrproject/zephyr/include/zephyr/net/net_compat.h:51:24: note: in expansion of macro 'NET_PF_INET'
   51 | #define AF_INET        NET_PF_INET
      |                        ^~~~~~~~~~~
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:442:30: note: in expansion of macro 'AF_INE'
  442 |                 .ai_family = AF_INET,
      |                              ^~~~~~~
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:443:18: error: 'const struct addrinfo' has no member named 'ai_socktype'
  443 |                 .ai_socktype = SOCK_STREAM
      |                  ^~~~~~~~~~~
In file included from T:/Dev/zephyrproject/zephyr/include/zephyr/net/net_ip.h:2223:
T:/Dev/zephyrproject/zephyr/include/zephyr/net/net_compat.h:76:21: warning: excess elements in struct initializer
   76 | #define SOCK_STREAM NET_SOCK_STREAM
      |                     ^~~~~~~~~~~~~~~
T:/Dev/zephyrproject/zephyr/include/zephyr/net/net_compat.h:76:21: note: in definition of macro 'SOCK_STREAM'
   76 | #define SOCK_STREAM NET_SOCK_STREAM
      |                     ^~~~~~~~~~~~~~~
T:/Dev/zephyrproject/zephyr/include/zephyr/net/net_compat.h:76:21: note: (near initialization for 'hints')
   76 | #define SOCK_STREAM NET_SOCK_STREAM
      |                     ^~~~~~~~~~~~~~~
T:/Dev/zephyrproject/zephyr/include/zephyr/net/net_compat.h:76:21: note: in definition of macro 'SOCK_STREAM'
   76 | #define SOCK_STREAM NET_SOCK_STREAM
      |                     ^~~~~~~~~~~~~~~
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:441:31: error: storage size of 'hints' isn't known
  441 |         const struct addrinfo hints = {
      |                               ^~~~~
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:447:14: warning: implicit declaration of function 'getaddrinfo'; did you mean 'net_addr_ntop'? [-Wimplicit-function-declaration]
  447 |         rc = getaddrinfo(CONFIG_NET_SAMPLE_MQTT_BROKER_HOSTNAME,
      |              ^~~~~~~~~~~
      |              net_addr_ntop
In file included from T:/Dev/zephyrproject/zephyr/include/zephyr/logging/log.h:11,
                 from T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:7:
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:450:67: warning: implicit declaration of function 'gai_strerror'; did you mean 'strerror'? [-Wimplicit-function-declaration]
  450 |                 LOG_ERR("Failed to resolve broker hostname [%s]", gai_strerror(rc));
      |                                                                   ^~~~~~~~~~~~
T:/Dev/zephyrproject/zephyr/include/zephyr/logging/log_core.h:223:32: note: in definition of macro 'Z_LOG_TO_PRINTK'
  223 |                              ##__VA_ARGS__); \
      |                                ^~~~~~~~~~~
T:/Dev/zephyrproject/zephyr/include/zephyr/logging/log_core.h:320:44: note: in expansion of macro 'Z_LOG2'
  320 | #define Z_LOG(_level, ...)                 Z_LOG2(_level, 0, Z_LOG_CURRENT_DATA(), __VA_ARGS__)
      |                                            ^~~~~~
T:/Dev/zephyrproject/zephyr/include/zephyr/logging/log.h:62:22: note: in expansion of macro 'Z_LOG'
   62 | #define LOG_ERR(...) Z_LOG(LOG_LEVEL_ERR, __VA_ARGS__)
      |                      ^~~~~
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:450:17: note: in expansion of macro 'LOG_ER'
  450 |                 LOG_ERR("Failed to resolve broker hostname [%s]", gai_strerror(rc));
      |                 ^~~~~~~
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:450:25: warning: format '%s' expects argument of type 'char *', but argument 2 has type 'int' [-Wformat=]
  450 |                 LOG_ERR("Failed to resolve broker hostname [%s]", gai_strerror(rc));
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~
      |                                                                   |
      |                                                                   int
T:/Dev/zephyrproject/zephyr/include/zephyr/logging/log_core.h:315:50: note: in definition of macro 'Z_LOG2'
  315 |                         z_log_printf_arg_checker(__VA_ARGS__);                                     \
      |                                                  ^~~~~~~~~~~
T:/Dev/zephyrproject/zephyr/include/zephyr/logging/log.h:62:22: note: in expansion of macro 'Z_LOG'
   62 | #define LOG_ERR(...) Z_LOG(LOG_LEVEL_ERR, __VA_ARGS__)
      |                      ^~~~~
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:450:17: note: in expansion of macro 'LOG_ER'
  450 |                 LOG_ERR("Failed to resolve broker hostname [%s]", gai_strerror(rc));
      |                 ^~~~~~~
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:450:62: note: format string is defined here 
  450 |                 LOG_ERR("Failed to resolve broker hostname [%s]", gai_strerror(rc));
      |                                                             ~^
      |                                                              |
      |                                                              char *
      |                                                             %d
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:459:65: error: invalid use of undefined type 'struct addrinfo'
  459 |         broker4->sin_addr.s_addr = ((struct sockaddr_in *)result->ai_addr)->sin_addr.s_addr;
      |                                                                 ^~
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:461:58: error: invalid use of undefined type 'struct addrinfo'
  461 |         broker4->sin_port = ((struct sockaddr_in *)result->ai_addr)->sin_port;
      |                                                          ^~
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:462:9: warning: implicit declaration of function 'freeaddrinfo'; did you mean 'zsock_freeaddrinfo'? [-Wimplicit-function-declaration]
  462 |         freeaddrinfo(result);
      |         ^~~~~~~~~~~~
      |         zsock_freeaddrinfo
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:465:9: warning: implicit declaration of function 'inet_ntop'; did you mean 'net_ntohl'? [-Wimplicit-function-declaration]
  465 |         inet_ntop(AF_INET, &broker4->sin_addr.s_addr, broker_ip, sizeof(broker_ip));
      |         ^~~~~~~~~
      |         net_ntohl
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:441:31: warning: unused variable 'hints' [-Wunused-variable]
  441 |         const struct addrinfo hints = {
      |                               ^~~~~
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c: At top level:
T:/Dev/zephyrproject/zephyr/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:30:22: warning: 'fds' defined but not used -Wunused-variable]
   30 | static struct pollfd fds[1];
      |                      ^~~
[102/401] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/assert.c.obj
ninja: build stopped: subcommand failed.
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

Prepare the environment for Ubuntu as described in https://docs.zephyrproject.org/latest/develop/getting_started/index.html (tested with upstream versions - last commit tested was 32823f39fbf2663fa1bbb75e60e5743cd6d91579)

### Additional Context

_No response_

--- a/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c
+++ b/samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c
@@ -7,10 +7,13 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(app_mqtt, LOG_LEVEL_DBG);
 
-#include <zephyr/kernel.h>
-#include <zephyr/net/socket.h>
-#include <zephyr/net/mqtt.h>
 #include <zephyr/data/json.h>
+#include <zephyr/kernel.h>
+#include <zephyr/net/mqtt.h>
+#include <zephyr/posix/arpa/inet.h>
+#include <zephyr/posix/netdb.h>
+#include <zephyr/posix/poll.h>
+#include <zephyr/posix/sys/socket.h>
 #include <zephyr/random/random.h>
 
 #include "mqtt_client.h"

